lib/plugins/aws/metrics/awsMetrics.js

```diff
@@ -55,160 +53,178 @@ class AwsMetrics {
+          'getMetricStatistics',
+          {
+            StartTime,
+            EndTime,
+            MetricName: 'Duration',
+            Namespace,
+            Period,
+            Dimensions: [
+              {
+                Name: 'FunctionName',
+                Value: FunctionName,
+              },
+            ],
+            Statistics: [
+              'Average',
+            ],
+            Unit: 'Milliseconds',
+          },
+          this.options.stage,
+          this.options.region
+        );
+
+      // push all promises to the array which will be used to resolve those
+      promises.push(invocationsPromise);
+      promises.push(throttlesPromise);
+      promises.push(errorsPromise);
+      promises.push(avgDurationPromise);
+
+      return BbPromise.all(promises).then((metrics) => metrics);
+    });
   }
 
   showMetrics(metrics) {
     let message = '';
 
-    message += `${chalk.yellow.underline(this.options.function)}\n`;
+    if (this.options.function) {
+      message += `${chalk.yellow.underline(this.options.function)}\n`;
+    } else {
+      message += `${chalk.yellow.underline('Service wide metrics')}\n`;
+    }
 
     const formattedStartTime = moment(this.options.startTime).format('LLL');
     const formattedEndTime = moment(this.options.endTime).format('LLL');
     message += `${formattedStartTime} - ${formattedEndTime}\n\n`;
 
     if (metrics && metrics.length > 0) {
+      let invocations = 0;
+      let throttles = 0;
+      let errors = 0;
+      let duration = 0;
+
       _.forEach(metrics, (metric) => {
-        if (metric.Label === 'Invocations') {
-          const datapoints = metric.Datapoints;
-          const invocations = datapoints
-            .reduce((previous, datapoint) => previous + datapoint.Sum, 0);
-          message += `${chalk.yellow('Invocations:', invocations, '\n')}`;
-        } else if (metric.Label === 'Throttles') {
-          const datapoints = metric.Datapoints;
-          const throttles = datapoints
-            .reduce((previous, datapoint) => previous + datapoint.Sum, 0);
-          message += `${chalk.yellow('Throttles:', throttles, '\n')}`;
-        } else if (metric.Label === 'Errors') {
-          const datapoints = metric.Datapoints;
-          const errors = datapoints
-            .reduce((previous, datapoint) => previous + datapoint.Sum, 0);
-          message += `${chalk.yellow('Errors:', errors, '\n')}`;
-        } else {
-          const datapoints = metric.Datapoints;
-          const duration = datapoints
-            .reduce((previous, datapoint) => previous + datapoint.Average, 0);
-          const formattedRoundedAvgDuration = `${Math.round(duration * 100) / 100}ms`;
-          message += `${chalk.yellow('Duration (avg.):', formattedRoundedAvgDuration)}`;
-        }
+        _.forEach(metric, (funcMetric) => {
+          if (funcMetric.Label === 'Invocations') {
+            invocations += utils.reduceDatapoints(funcMetric.Datapoints, 'Sum');
```

Comment:
hmm, not sure if I would put that into utils. Usage is pretty limited to this plugin. I think it might be move there once it's needed in two different plugins/places. For now I would leave it in this file.

It's always an issue with "utils" libs that they become garbage eventually.